src/main/java/sirius/biz/util/ArchiveExtractor.java

```diff
@@ -0,0 +1,313 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.util;
+
+import net.sf.sevenzipjbinding.ArchiveFormat;
+import net.sf.sevenzipjbinding.IInArchive;
+import net.sf.sevenzipjbinding.SevenZip;
+import net.sf.sevenzipjbinding.SevenZipException;
+import net.sf.sevenzipjbinding.impl.RandomAccessFileInStream;
+import sirius.biz.storage.layer3.VirtualFile;
+import sirius.kernel.Sirius;
+import sirius.kernel.commons.Amount;
+import sirius.kernel.commons.Callback;
+import sirius.kernel.commons.Explain;
+import sirius.kernel.commons.Files;
+import sirius.kernel.commons.Processor;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Exceptions;
+import sirius.kernel.health.Log;
+import sirius.kernel.nls.NLS;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.RandomAccessFile;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Enumeration;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipException;
+import java.util.zip.ZipFile;
+
+/**
+ * Utility to handle and extract archive files.
+ * <p>
+ * If the framework <tt>biz.seven-zip</tt> is enabled, this will load the native 7-ZIP bindings and support a
+ * multitude of file formats like <tt>7z, tar, rar etc.</tt> otherwise "just" ZIP files are supported.
+ * <p>
+ * Note that using native code always bears a certain risk as an error there crashed the whole JVM. Therefore enabling
+ * this is only advised on clusters which have dedicated worker servers.
+ * <p>
+ * Also note that ZIP files are always processed using the Java APIs as this is way simpler to process (can be
+ * done in memory without allocating large buffers or resorting to disk - therefore this is also faster (as disk
+ * access is always slower than executing Java-Code). Also 7-ZIP has some severe bugs when handling paths in ZIP
+ * files which were produced by Microsoft Windows.
+ * <p>
+ * Note that {@link #getSupportedFileExtensions()} honors the framework selection and always returns an appropriate
+ * list.
+ */
+@Register(classes = ArchiveExtractor.class)
+public class ArchiveExtractor {
```

Comment:
Is this a replacement for the ArchiveHelper class?

Reply:
yes this will be deleted in a subsequent PR